test/rethinkdb/joins_test.clj

```diff
@@ -0,0 +1,51 @@
+(ns rethinkdb.joins-test
+  (:require [clojure.test :refer :all]
+            [rethinkdb.query :as r]
+            [rethinkdb.test-utils :as u]))
+
+(def names [{:id 1 :name "Pikachu", :type 2},
+            {:id 2 :name "Magnemite", :type [1 2]}])
```

Comment:
Seeing Magnemite with two types here suggests that it will be somehow used in the joins below. It would be good to add a comment either here or on the joins, explaining why it doesn't show up.